src/CppUTest/CommandLineTestRunner.cpp

```diff
@@ -51,21 +49,44 @@ int CommandLineTestRunner::RunAllTests(int ac, char** av)
 int CommandLineTestRunner::RunAllTests(int ac, const char** av)
 {
     int result = 0;
-    ConsoleTestOutput output;
+    TestRegistry* registry = TestRegistry::getCurrentRegistry();
+    CommandLineArguments arguments( ac, av );
+    arguments.parse( registry->getFirstPlugin() );
+    TestOutput* output;
+    if( arguments.isJUnitOutput() ) {
+        JUnitTestOutput* jUnitOutput = new JUnitTestOutput;
+        jUnitOutput->setPackageName( arguments.getPackageName() );
+        output = jUnitOutput;
+    }
+    else {
+        output = new ConsoleTestOutput;
+    }
 
     MemoryLeakWarningPlugin memLeakWarn(DEF_PLUGIN_MEM_LEAK);
     memLeakWarn.destroyGlobalDetectorAndTurnOffMemoryLeakDetectionInDestructor(true);
-    TestRegistry::getCurrentRegistry()->installPlugin(&memLeakWarn);
+    registry->installPlugin( &memLeakWarn );
+
+    SetPointerPlugin pPlugin( DEF_PLUGIN_SET_POINTER );
+    registry->installPlugin( &pPlugin );
 
+    if(arguments.isSuccesfullyParsed())
     {
-        CommandLineTestRunner runner(ac, av, &output, TestRegistry::getCurrentRegistry());
+        CommandLineTestRunner runner( output, registry, &arguments );
         result = runner.runAllTestsMain();
     }
+    else {
+        output->print( arguments.usage() );
+    }
+
+    registry->removePluginByName( DEF_PLUGIN_SET_POINTER );
 
     if (result == 0) {
-        output << memLeakWarn.FinalReport(0);
+        *output << memLeakWarn.FinalReport(0);
     }
-    TestRegistry::getCurrentRegistry()->removePluginByName(DEF_PLUGIN_MEM_LEAK);
+    registry->removePluginByName( DEF_PLUGIN_MEM_LEAK );
+
+    delete output;
+
```

Comment:
It sounds sensible and the moment the 3rd step of the decorator I think I see again how that is possible but I did not know how to do it now. Well, maybe with that if else within the if succesfullyParsed block and then create the testrunner with the one or the other.